### PR TITLE
Enum switchers

### PR DESCRIPTION
**Pull request by xarvic**
Add enum-switchers similar to [`druid enums`](https://github.com/Finnerale/druid-enums/blob/master/src/lib.rs), which uses prisms instead.

### CODE REVIEW

**Comment by derekdreery** [comment]
I'm excited about trying this out, but I merged some other patches first and they conflict. Would you mind resolving the conflicts? You might also want to look at [`druid-enums`](https://github.com/Finnerale/druid-enums) which allows you to build a view switcher for variants of an enum.

**Comment by xarvic** [comment]
I know druid-enums. My plan was to write a comparison of druid-enums and my aproach using prims in #1135 when this PR is merged.
I would like to resolve the conflicts, but I don't have write permission.

**Comment by derekdreery** [comment]
Don't you pull the changes to your computer then merge or rebase? That's
how I've always done it. I think more people should have write permissions
need to come up with a plan for that, maybe 1 review like other druid repos

**Comment by djeedai** [comment]
Not sure why GitHub asks write permission to resolve conflicts on own PR, but it's probably better to do it locally anyway to make sure things still work, given CI is red and won't catch any error.

**Comment by xarvic** [comment]
> Don't you pull the changes to your computer then merge or rebase? That's
how I've always done it. I think more people should have write permissions
need to come up with a plan for that, maybe 1 review like other druid repos

I always used Github's mergetool, but i can also do it that way :)

File: examples/partial.rs [hunk 1]
```diff
@@ -16,23 +15,25 @@ fn main_widget() -> impl Widget<TestData> {
...
-    //         }
-    //     },
-    //     TestData::B,
-    // );
+    let partial = DisablePrismWrap::new(
+        TextBox::new(),
+        String::new(),
+        Closures(
+            |outer: &TestData| {
+                if let TestData::B(str) = outer {
```
Comment (djeedai) [inline examples/partial.rs:23]: Minor suggestion: avoid using `str` as variable name since it's also the name of a type (string slice `&str`).

**Review by djeedai — APPROVED** [review]

File: examples/partial.rs (hunk 1 above)
Comment (xarvic) [reply examples/partial.rs]: Ok, i changed it

**Comment by derekdreery** [comment]
Thanks for this. Let's get it in-tree and experiment there.

**Pull request merged**